src/Simulation/EntryPointDriver/Azure/Azure.cs

```diff
@@ -0,0 +1,308 @@
+        /// <param name="machine">The quantum machine used for submission.</param>
+        /// <param name="submission">The Q# entry point submission.</param>
+        /// <returns>The exit code.</returns>
+        private static async Task<int> SubmitQSharpMachine<TIn, TOut>(
+            AzureSettings settings, IQuantumMachine machine, QSharpSubmission<TIn, TOut> submission)
+        {
+            Log(settings, "Submitting Q# entry point using a quantum machine.");
+
+            if (settings.DryRun)
+            {
+                var (valid, message) = machine.Validate(submission.EntryPointInfo, submission.Argument);
+                return DisplayValidation(valid ? null : message);
+            }
+
+            var job = machine.SubmitAsync(
+                submission.EntryPointInfo,
+                submission.Argument,
+                new SubmissionContext { FriendlyName = settings.JobName, Shots = settings.Shots });
+
+            return await DisplayJobOrError(settings, job);
+        }
+
+        /// <summary>
+        /// Submits a Q# entry point to Azure Quantum.
+        /// </summary>
+        /// <typeparam name="TIn">The entry point's argument type.</typeparam>
+        /// <typeparam name="TOut">The entry point's return type.</typeparam>
+        /// <param name="settings">The Azure submission settings.</param>
+        /// <param name="submitter">The Q# submitter.</param>
+        /// <param name="submission">The Q# entry point submission.</param>
+        /// <returns>The exit code.</returns>
+        private static async Task<int> SubmitQSharp<TIn, TOut>(
+            AzureSettings settings, IQSharpSubmitter submitter, QSharpSubmission<TIn, TOut> submission)
+        {
+            Log(settings, "Submitting Q# entry point.");
+
+            if (settings.DryRun)
+            {
+                return DisplayValidation(submitter.Validate(submission.EntryPointInfo, submission.Argument));
+            }
+
+            var job = submitter.SubmitAsync(
+                submission.EntryPointInfo, submission.Argument, settings.SubmissionOptions);
+            return await DisplayJobOrError(settings, job);
+        }
+
+        /// <summary>
+        /// Submits a QIR entry point to Azure Quantum. 
+        /// </summary>
+        /// <param name="settings">The Azure submission settings.</param>
+        /// <param name="submitter">The QIR entry point submitter.</param>
+        /// <param name="submission">The QIR entry point submission.</param>
+        /// <returns>The exit code.</returns>
+        private static async Task<int> SubmitQir(
+            AzureSettings settings, IQirSubmitter submitter, QirSubmission submission)
+        {
+            Log(settings, "Submitting QIR entry point.");
+
+            if (settings.DryRun)
+            {
+                DisplayError("Dry run is not supported with QIR submission.", null);
+                return 1;
```

Comment:
Should we have constants for specific exit codes?

Reply:
I believe it might be worth doing when we have a larger set of error conditions but for now I believe it is ok to return the same exit code regardless of the specific error.